shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/api/impl/RuleAlteredJobAPIImpl.java

```diff
@@ -314,18 +320,20 @@ public void switchClusterConfiguration(final JobConfiguration jobConfig) {
                 throw new PipelineDataConsistencyCheckFailedException("Data consistency check not finished or failed.");
             }
         }
-        Optional<Collection<RuleAlteredJobContext>> optionalJobContexts = RuleAlteredJobSchedulerCenter.getJobContexts(jobId);
-        optionalJobContexts.ifPresent(jobContexts -> jobContexts.forEach(each -> each.setStatus(JobStatus.ALMOST_FINISHED)));
```

Comment:
`JobStatus.ALMOST_FINISHED` is used in `RuleAlteredJobScheduler.stop()` to release resources via `jobPreparer.cleanup` for PostgreSQL and openGauss.
Related code:
```
        final boolean almostFinished = jobContext.getStatus() == JobStatus.ALMOST_FINISHED;
        if (almostFinished) {
            log.info("almost finished, preparer cleanup, job {}", jobContext.getJobId());
            RuleAlteredJobPreparer jobPreparer = jobContext.getJobPreparer();
            if (null != jobPreparer) {
                jobPreparer.cleanup(jobContext);
            }
        }
```

Looks `JobStatus.ALMOST_FINISHED` could not be fetched from `jobContext.getStatus()` any more after modification.